### PR TITLE
docs: add recommended dev setup

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR adds a new repo-level documentation page: `DEV_SETUP.md`. It contains detailed and "officially recommended" setup guides for getting a machine ready for Opentrons development.

As of this PR, I've only fully fleshed out the macOS section, which in my testing works on both Intel and M1 macs. I've left outlines for Linux and Windows based on setups I have used in the past on my machines. This guide is built from a lot of research and testing by other people (including but not limited to @y3rsh, @TamarZanzouri, and @koji) and is a codification of their findings.

**NOTE:** instructions as written may not work fully on M1 macs until #9880 is merged

## Changelog

- Add `DEV_SETUP.md` with development machine setup guide

## Review requests

- Do these instructions work for you on a fresh machine?
- Does this filename / location make sense? If not, any other ideas or proposals?
    - I'd also be down for a `dev-docs` directory or something 

## Risk assessment

Low. Documentation.

### CODE REVIEW

File: DEV_SETUP.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+echo 'eval "$(pyenv init -)"' >> ~/.zshrc
+```
+
+Close and re-open your terminal to verify that `pyenv` is installed
+
+```shell
+pyenv --version
+```
+
+Now, install the required version of Python. Use the latest available version of `3.7.x`, which is `3.7.13` at the time or writing.
```
Comment (mcous) [inline DEV_SETUP.md]:
```suggestion
Now, install the required version of Python. Use the latest available version of `3.7.x`, which is `3.7.13` at the time of writing.
```

**Comment by alexjoel42** [comment]
Hi! 

FWIW, I wanted to add some of my feedback from working with the developer environment.

 When I was struggling with this [a few months ago](https://github.com/Opentrons/opentrons/commit/32ea047c40c5566bdbcec41651a5d187455c520d) I tried to write down some additional steps 


  **MacOS M1 Chip Note** please install xcode tools as well as [rosetta 2](https://support.apple.com/en-us/HT211861) in order to properly access the functions. 
  ```shell
  xcode-select --install
  ```
  **Windows Note**  nvm doesn't support windows. You'd have to go download [node.js](https://nodejs.org/en/download/) and then install it using [chocolatey](https://classic.yarnpkg.com/en/docs/install#windows-stable).
  ```shell
  choco install yarn
  ```

**Comment by mcous** [comment]
Hey @alexjoel42, I believe these instructions address your points

> **MacOS M1 Chip Note** please install xcode tools as well as [rosetta 2](https://support.apple.com/en-us/HT211861) in order to properly access the functions.

- These instructions walk you through installing Homebrew, which will install the XCode Command Line tools as part of its installation process
- **Rosetta is not required** for this setup, which is why these docs don't call it out

> Windows Note nvm doesn't support windows. You'd have to go download [node.js](https://nodejs.org/en/download/) and then install it using [chocolatey](https://classic.yarnpkg.com/en/docs/install#windows-stable).

These docs recommend **[Node Version Switcher aka `nvs`](https://github.com/jasongin/nvs)**. Unlike nv**m**, nv**s** is **fully supported on Windows**. I've been using `nvs` on both mac and windows for a few years now, and it works great.

Likewise, these docs (or rather, the outline so far) recommend `scoop` over `chocolatey` on Windows. On my Windows machines, `scoop` has been a lot easier to use than Chocolatey for installing different versions of Python.

File: DEV_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+- [yarn] to manage JavaScript depenencies
+- [pipenv] to manage Python dependencies
+
+```shell
+# install yarn v1
+npm install --global yarn@1
+
+# install pipenv v2021.5.29
+pip install pipenv==2021.5.29
```
Comment (y3rsh) [inline DEV_SETUP.md]: This is already a part of [`setup-py`](https://github.com/Opentrons/opentrons/blob/2d4f04429844375b3e76abd499b555db5fae0a8a/Makefile#L60) in our makefile.  Also needed here?

**Review by y3rsh — APPROVED** [review]
Just one question.  Looks great!  Huge improvement.

File: DEV_SETUP.md (hunk 2 above)
Comment (mcous) [reply DEV_SETUP.md]: Ah good call, might not be. Will test and remove if redundant

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -442,7 +394,7 @@ Our systemd units are:
 
 #### Buildroot
 
-When a robot is running on buildroot, its filesystem is mounted from two separate locations. `/data`, `/var`, and `/home` are from the "data" partition, and everything else is from the root partition (or generated by the system). The root partition is what gets updated, by being overwritten. To make this work, the root partition is mounted readonly, which causes writes to files in that partition to fail with the error "readonly filesystem". To prevent this, you can remount the partition:
+An OT-2's filesystem is mounted from two separate locations. `/data`, `/var`, and `/home` are from the "data" partition, and everything else is from the root partition (or generated by the system). The root partition is what gets updated, by being overwritten. To make this work, the root partition is mounted readonly, which causes writes to files in that partition to fail with the error "readonly filesystem". To prevent this, you can remount the partition:
```
Comment (SyntaxColoring) [inline CONTRIBUTING.md:411]: Do you want to delete the `#### Buildroot` subheading here, too? You've deleted the other `#### Buildroot` subheadings.

**Review by SyntaxColoring — APPROVED** [review]
Looks great. Thanks!

File: CONTRIBUTING.md (hunk 3 above)
Comment (mcous) [reply CONTRIBUTING.md:411]: Oops, yeah meant to do that

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -401,37 +387,19 @@ make term br_ssh_key=/path/to/privkey
...
+OT-2 robots use [systemd-journald][] for log management. This is a single log manager for everything on the system. It is administrated using the [journalctl][] utility. You can view logs by just doing `journalctl` (it may be better to do `journalctl --no-pager | less` to get a better log viewer), or stream them by doing `journalctl -f`. Any command that displays logs can be narrowed down by using a syslog identifier: `journalctl -f SYSLOG_IDENTIFIER=opentrons-api` will only print logs from the api server's loggers, for instance. Our syslog identifiers are:
 
 - `opentrons-api`: The API server - anything sent to `logging` logs from the api server package, except the serial logs
-- `opentrons-update-server`: Anything sent to `logging` logs from the update server packate
+- `opentrons-update-server`: Anything sent to `logging` logs from the update server package
 - `opentrons-api-serial`: The serial logs
```
Comment (sanni-t) [inline CONTRIBUTING.md:398]: Should we include `uvicorn` for logs of http requests?

**Review by sanni-t — APPROVED** [review]

**Review by aptrons22 — COMMENTED** [review]
Hello Mike,
The revisions for the M1 specs are great.  Thanks for revising.

**Review by aptrons22 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (mcous) [reply CONTRIBUTING.md:398]: Yeah, let's do it in a follow-up so we can concentrate on getting it correct

**Pull request merged**